pkg/overlay/overlay.go

```diff
@@ -61,23 +75,32 @@ func generateOverlayStructure(containerDir string, rootUID, rootGID int) (string
 	return containerDir, nil
 }
 
+// MountWithOptions creates a subdir of the contentDir based on the source directory
+// from the source system.  It then mounts up the source directory on to the
+// generated mount point and returns the mount point to the caller.
+// But allows api to set custom workdir, upperdir and other overlay options
+// Following API is being used by podman at the moment
+func MountWithOptions(contentDir, source, dest string, rootUID, rootGID int, opts *Options) (mount specs.Mount, Err error) {
+	return mountHelper(contentDir, source, dest, rootUID, rootGID, opts.Graphopts, opts, opts.Readonly)
+}
+
 // Mount creates a subdir of the contentDir based on the source directory
 // from the source system.  It then mounts up the source directory on to the
 // generated mount point and returns the mount point to the caller.
 func Mount(contentDir, source, dest string, rootUID, rootGID int, graphOptions []string) (mount specs.Mount, Err error) {
-	return mountHelper(contentDir, source, dest, rootUID, rootGID, graphOptions, false)
+	return mountHelper(contentDir, source, dest, rootUID, rootGID, graphOptions, nil, false)
 }
 
 // MountReadOnly creates a subdir of the contentDir based on the source directory
 // from the source system.  It then mounts up the source directory on to the
 // generated mount point and returns the mount point to the caller.  Note that no
 // upper layer will be created rendering it a read-only mount
 func MountReadOnly(contentDir, source, dest string, rootUID, rootGID int, graphOptions []string) (mount specs.Mount, Err error) {
-	return mountHelper(contentDir, source, dest, rootUID, rootGID, graphOptions, true)
+	return mountHelper(contentDir, source, dest, rootUID, rootGID, graphOptions, nil, true)
 }
 
 // NOTE: rootUID and rootUID are not yet used.
-func mountHelper(contentDir, source, dest string, _, _ int, graphOptions []string, readOnly bool) (mount specs.Mount, Err error) {
+func mountHelper(contentDir, source, dest string, _, _ int, graphOptions []string, opts *Options, readOnly bool) (mount specs.Mount, Err error) {
```

Comment:
It’s now surprising for `opts.Readonly` and `opts.Graphopts` to be passed to this function, but ignored. I think the individual `graphOptions`, `readOnly` parameters to this function should be dropped (and the two old public functions would be passing them via a single-use `Options` value).

Reply:
@mtrmac `MountWithOptions` automatically sets these but I agree this is making things confusing for reader. Let me remove `mountHelper` that should simplify things.  https://github.com/containers/buildah/pull/3715/files#diff-97a773bb4375bb0de346986ec764abdbc2f07b74a298953de1973eb747ed9759R84

Reply:
This is fixed in latest commit.